src/containers/pages/InterventionPlan/IRS/plan/serviceCalls.ts

```diff
@@ -0,0 +1,31 @@
+import { OPENSRP_PLANS } from '../../../../../constants';
+import { displayError } from '../../../../../helpers/errors';
+import { OpenSRPService } from '../../../../../services/opensrp';
+import {
+  extractPlanRecordResponseFromPlanPayload,
+  fetchPlanRecords,
+  PlanPayload,
+} from '../../../../../store/ducks/plans';
+
+type UUID = string;
+
+/** loads a plan and dispatches action to add plan to store
+ * @param {OpenSRPService} service - the opensrp service
+ * @param {UUID} planId - uuid of the plan to be got in the api call
+ * @param {actionCreator} - action creator.
+ */
+export async function loadPlan(
+  service: typeof OpenSRPService,
+  planId: UUID,
+  actionCreator: typeof fetchPlanRecords
+) {
+  const OpenSrpPlanService = new service(OPENSRP_PLANS);
+  await OpenSrpPlanService.read(planId)
+    .then((plan: PlanPayload[]) => {
+      const planRecord = extractPlanRecordResponseFromPlanPayload(plan[0]);
+      if (planRecord) {
+        actionCreator([planRecord]);
+      }
```

Comment:
What happens when `planRecord` is false-ey?  Can we call displayError with some error message?